src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,326 @@
+            return new IncorrectCommand("The description of a todo cannot be empty.");
+        } else {
+            return new AddCommand(
+                    new Todo(request
+                            .substring(5)
+                            .strip()));
+        }
+    }
+
+    /**
+     * Parses input in the context of adding a deadline.
+     *
+     * @param request full user input
+     * @return the prepared add command
+     */
+    private Command prepareDeadline(String request)  {
+        if (request.strip()
+                    .length() == 8) {
+            return new IncorrectCommand("The description of a deadline cannot be empty.");
+        } else if (!request.contains(" /by ")) {
+            return new IncorrectCommand("You left the date/time of the deadline empty!");
+        }
+
+        String next = request.substring(8);
+        String[] parsedReq = next.split("/by");
+        String desc = parsedReq[0].strip();
+        String by = parsedReq[1].strip();
+
+        if (desc.length() == 0) {
+            return new IncorrectCommand("The description of a deadline cannot be empty.");
+        } else if (by.length() == 0) {
+            return new IncorrectCommand("You left the date/time of the deadline empty!");
+        } else {
+            Parser.Format f = Parser.parseDateTime(by);
+
+            switch (f) {
+            case DATETIME:
+                String[] s = by.split(" ");
+                return new AddCommand(
+                        new Deadline(desc,
+                                LocalDate.parse(s[0], dateIn),
+                                LocalTime.parse(s[1], timeIn)));
+            case DATE:
+                return new AddCommand(
+                        new Deadline(desc,
+                                LocalDate.parse(by, dateIn)));
+            case TIME:
+                return new AddCommand(
+                        new Deadline(desc,
+                                LocalTime.parse(by, timeIn)));
+            case INVALID:
+            default:
+                return new IncorrectCommand("Please enter the date and/or time in the specified format:\n" +
+                        "    yyyy-MM-dd HHmm\n" +
+                        "    yyyy-MM-dd\n" +
+                        "    or HHmm");
+            }
+        }
+    }
+
+    /**
+     * Parses input in the context of adding an event.
+     *
+     * @param request full user input
+     * @return the prepared add command
+     */
+    private Command prepareEvent(String request) {
+        if (request.strip()
+                    .length() == 5) {
+            return new IncorrectCommand("The description of an event cannot be empty.");
+        } else if (!request.contains(" /at ")) {
+            return new IncorrectCommand("You left the date/time of the event empty!");
+        }
+
+        String next = request.substring(5);
+        String[] parsedReq = next.split("/at");
+        String desc = parsedReq[0].strip();
```

Comment:
A bit ambiguous to this variable name desc, maybe you can refactor its name to description